services-core/src/gas_price/priority.rs

```diff
@@ -17,10 +39,21 @@ impl PriorityGasPrice {
         T: Fn(&'a dyn GasPriceEstimating) -> F,
         F: Future<Output = Result<f64>>,
     {
-        for estimator in &self.estimators {
-            match operation(estimator.as_ref()).await {
-                Ok(result) => return Ok(result),
-                Err(err) => log::error!("gas estimator failed: {:?}", err),
+        const ORDERING: Ordering = Ordering::SeqCst;
+        for (i, estimator) in self.estimators.iter().enumerate() {
+            match operation(estimator.estimator.as_ref()).await {
+                Ok(result) => {
+                    estimator.errors_in_a_row.store(0, ORDERING);
+                    return Ok(result);
+                }
+                Err(err) => {
+                    let num_errors = estimator.errors_in_a_row.fetch_add(1, ORDERING) + 1;
```

Comment:
Are you are adding 1 twice here? It seems like after the first error num_errors would evaluate to 2.

Reply:
`fetch_add` returns the previous value (`fetch` first then `add`). I *think* it is correct like this.